File: webclient/javascript/house.js
class House {
  constructor(position, owner) {
    this.position = position;
    this.owner = owner;
  }

  serialize() {
    return {
      position: this.position,
      owner: this.owner
    }
  }

  static deserialize(house) {
    return new House(house.position, house.owner);
  }

  //TODO Consider a baseclass for these
  static deserializeList(houses) {
    return houses.map(house => {
      return House.deserialize(house);
    });
  }
}

export default House;

File: server/models/house.py
class House:
    def __init__(self, house_id, position, owner):
        self.id = house_id 
        self.position = position
        self.owner = owner

    def pack(self):
        return {
            'id': self.id,
            'position': self.position,
            'owner': self.owner,
        }
